docs/build/sdks/python/copilot.mdx

```diff
@@ -0,0 +1,256 @@
+
+# Ask the question and stream the response
+for response in pieces_client.copilot.stream_question(question):
+   if response.question:
+         # Each answer is a chunk of the entire response to the question
+         answers = response.question.answers.iterable
+         for answer in answers:
+            print(answer.text,end="")
+
+# Close the client
+pieces_client.close()
+```
+
+If you do not want to stream the response, you can use the `quesiton()` method instead.
+
+#### Example
+
+```python
+from pieces_os_client.wrapper import PiecesClient
+
+# Initialize the PiecesClient
+pieces_client = PiecesClient()
+
+# Set the question you want to ask
+question = "What is Object-Oriented Programming?"
+
+# Ask the question and get the response
+response = pieces_client.copilot.question(question)
+
+# Print the response message
+message = response.answers.iterable[0].text
+print(message)
+
+# Close the client
+pieces_client.close()
+```
+
+### Get All Chats
+
+The `chats()` method returns a list of all chats.
+
+#### Example
+
+```python
+from pieces_os_client.wrapper import PiecesClient
+
+# Initialize the PiecesClient
+pieces_client = PiecesClient()
+
+# Get all chats and print their names
+chats = pieces_client.copilot.chats()
+for chat in chats:
+    print(chat.name)
+
+# Close the client
+pieces_client.close()
+```
+
+{/* need section on selecting an llm */}
+
+## LLM Management
+
+### Set the Current LLM
+
+The `model_name` setter method allows you to set the current LLM.
+
+#### Example
+
+```python
+from pieces_os_client.wrapper import PiecesClient
+
+# Initialize the PiecesClient
+pieces_client = PiecesClient()
+
+# Set the current LLM
+pieces_client.model_name = "your_model_name"
+
+# Close the client
+pieces_client.close()
+```
+
+### Get Availiable LLMs
+
+The `get_models()` method returns a list of available LLMs as a dictionary.
+
+#### Example
+
+```python
+from pieces_os_client.wrapper import PiecesClient
+from pieces_os_client import ClassificationSpecificEnum, FragmentMetadata
+
+# Initialize the PiecesClient
+pieces_client = PiecesClient()
+
+# Get all models and print their names
+models = pieces_client.get_models()
+for model_name, model_id in models.items():
+    print(model_name)
+
+# Close the client
+pieces_client.close()
+```
+
+### Copilot Class
+
+The `Copilot` class provides a way to manage copilot functionality with various properties and methods.
+
+#### Properties
+
+| Property Name | Property Description |
+|--------------- |----------------------|
+| **chat**      | The chat object. |
+
+#### Methods
+
+##### Set the Current Chat
+
+The `chat` setter method allows you to set the current chat.
+
+:::info
+
+Changing the current chat will clear all context added to the chat.
+
+:::
+
+#### Example
+
+```python
+from pieces_os_client.wrapper import PiecesClient
+
+# Initialize the PiecesClient
+pieces_client = PiecesClient()
+
+# Get the chat ID
+chat_id = "your_chat_id"
+
+# Set the current chat
```

Comment:
You should give the BasicChat object here not the id

Reply:
Can you give an example of this?

Reply:
pieces_client.copilot.chat=BasicChat(id)

Reply:
updated